device/tests/Microsoft.Azure.Devices.Client.Test/DeviceClientTests.cs

```diff
@@ -75,6 +75,22 @@ public void DeviceClient_SetDiagnosticSamplingPercentageOutOfRange_Fail()
             }
         }
 
+        [TestMethod]
+        [TestCategory("IoTHubClientDiagnostic")]
+        public void DeviceClient_StartDiagLocallyThatDoNotSupport_ThrowException()
+        {
+            DeviceClient deviceClient = DeviceClient.CreateFromConnectionString(fakeConnectionString, TransportType.Http1);
```

Comment:
Please add another test where multiple transport types are used: transport choices w/o Http1 should work while those containing Http1 anywhere in the list should fail.

Reply:
fixed